modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/SqlQueryType.java

```diff
@@ -33,7 +34,15 @@ public enum SqlQueryType {
     DDL,
 
     /** Explain. */
-    EXPLAIN;
+    EXPLAIN,
+
+    /** Transaction control statements such as {@code START TRANSACTION}, {@code COMMIT},  etc. */
+    TX_CONTROL,
+
+    ;
+
+    /** A set of statement types that can run only in single statement mode. **/
```

Comment:
For me this doesn't match with current PR description "Add restrictions for execution tx related statements with single statement mode" :thinking_face:

I thought that the main goal is to throw exception if SqlQueryProcessor#querySingle processing tx-related statements, but may be I missing something?

Reply:
@zstan 
can you clarify what the main goal of the ticket is?

Reply:
@xtern 
I think you missed tests cases in `testTxControlStatementsAreNotAllowdWithExplicitTransactions`

Reply:
Sorry, I missed "complementOf" :pensive: